src/Test.sol

```diff
@@ -139,6 +143,202 @@ abstract contract Test is DSTest {
+
+        uint256 percentDelta = ((a > b ? a - b : b - a) * 1e18) / b;
+
+        if (percentDelta > maxPercentDelta) {
+            emit log                    ("Error: a ~= b not satisfied [uint]");
+            emit log_named_uint         ("    Expected", b);
+            emit log_named_uint         ("      Actual", a);
+            emit log_named_decimal_uint (" Max % Delta", maxPercentDelta, 18);
+            emit log_named_decimal_uint ("     % Delta", percentDelta, 18);
+            fail();
+        }
+    }
+
+    function assertApproxEqRel(
+        uint256 a,
+        uint256 b,
+        uint256 maxPercentDelta, // An 18 decimal fixed point number, where 1e18 == 100%
+        string memory err
+    ) internal virtual {
+        if (b == 0) return assertEq(a, b); // If the expected is 0, actual must be too.
+
+        uint256 percentDelta = ((a > b ? a - b : b - a) * 1e18) / b;
+
+        if (percentDelta > maxPercentDelta) {
+            emit log_named_string       ("Error", err);
+            emit log_named_uint         ("    Expected", b);
+            emit log_named_uint         ("      Actual", a);
+            emit log_named_decimal_uint (" Max % Delta", maxPercentDelta, 18);
+            emit log_named_decimal_uint ("     % Delta", percentDelta, 18);
+            fail();
+        }
+    }
+
+    function assertApproxEqRel(
+        int256 a,
+        int256 b,
+        int256 maxPercentDelta // An 18 decimal fixed point number, where 1e18 == 100%
+    ) internal virtual {
+        if (b == 0) return assertEq(a, b); // If the expected is 0, actual must be too.
+
+        int256 percentDelta = ((a > b ? a - b : b - a) * 1e18) / b;
+
+        if (percentDelta > maxPercentDelta) {
+            emit log                   ("Error: a ~= b not satisfied [uint]");
+            emit log_named_int         ("    Expected", b);
+            emit log_named_int         ("      Actual", a);
+            emit log_named_decimal_int (" Max % Delta", maxPercentDelta, 18);
+            emit log_named_decimal_int ("     % Delta", percentDelta, 18);
+            fail();
+        }
+    }
+
+    function assertApproxEqRel(
+        int256 a,
+        int256 b,
+        int256 maxPercentDelta, // An 18 decimal fixed point number, where 1e18 == 100%
+        string memory err
+    ) internal virtual {
+        if (b == 0) return assertEq(a, b); // If the expected is 0, actual must be too.
+
+        int256 percentDelta = ((a > b ? a - b : b - a) * 1e18) / b;
+
+        if (percentDelta > maxPercentDelta) {
+            emit log_named_string      ("Error", err);
+            emit log_named_int         ("    Expected", b);
+            emit log_named_int         ("      Actual", a);
+            emit log_named_decimal_int (" Max % Delta", maxPercentDelta, 18);
+            emit log_named_decimal_int ("     % Delta", percentDelta, 18);
+            fail();
+        }
+    }
 }
 
 
```

Comment:
We should add a header here before `stdError` "STD-ERRORS" to separate them from the assertions.

Reply:
done, can resolve if happy